### PR TITLE
Add support for Maximum QoS property on CONNACK

### PR DESCRIPTION
**Pull request by nayato**
Adds support for Maximum QoS on CONNACK to comply with MQTT 5 specification.

### CODE REVIEW

**Comment by nayato** [comment]
looking at CI run, failures do not appear to be related to this PR

File: Source/MQTTnet/Client/Connecting/MqttClientAuthenticateResult.cs [hunk 1]
```diff
@@ -25,6 +26,8 @@ public class MqttClientAuthenticateResult
         public string ReasonString { get; set; }
 
         public ushort? ReceiveMaximum { get; set; }
+        
+        public MqttQualityOfServiceLevel? MaximumQoS { get; set; }
```
Comment (chkr1011) [inline Source/MQTTnet/Client/Connecting/MqttClientAuthenticateResult.cs:30]: In my opinion this should be not nullable. If the value is not present this should default to QoS 2 (From spec: Followed by a Byte with a value of either 0 or 1. It is a Protocol Error to include Maximum QoS more than once, or to have a value other than 0 or 1. If the Maximum QoS is absent, the Client uses a Maximum QoS of 2.). 

But not in this class. The conversion should be made in the data converter instead.

File: Source/MQTTnet/Formatter/V5/MqttV500DataConverter.cs [hunk 2]
```diff
@@ -93,6 +93,7 @@ public MqttClientAuthenticateResult CreateClientConnectResult(MqttConnAckPacket
                 MaximumPacketSize = connAckPacket.Properties?.MaximumPacketSize,
                 ReasonString = connAckPacket.Properties?.ReasonString,
                 ReceiveMaximum = connAckPacket.Properties?.ReceiveMaximum,
+                MaximumQoS = connAckPacket.Properties?.MaximumQoS,
                 ResponseInformation = connAckPacket.Properties?.ResponseInformation,
                 TopicAliasMaximum = connAckPacket.Properties?.TopicAliasMaximum,
```
Comment (chkr1011) [inline Source/MQTTnet/Formatter/V5/MqttV500DataConverter.cs:98]: As stated above just add _?? MqttQualityOfServiceLevel.ExactlyOnce_

File: Source/MQTTnet/Formatter/V5/MqttV500PropertiesReader.cs [hunk 3]
```diff
@@ -82,6 +82,16 @@ public ushort ReadReceiveMaximum()
...
         }
 
+        public MqttQualityOfServiceLevel ReadMaximumQoS()
+        {
+            byte value = _body.ReadByte();
+            if (value > 1)
+            {
+                throw new MqttProtocolViolationException($"Unexpected Maximum QoS value: {value}");
+            }
+            return (MqttQualityOfServiceLevel)value;
```
Comment (chkr1011) [inline Source/MQTTnet/Formatter/V5/MqttV500PropertiesReader.cs:92]: Please add empty line below the }.

**Review by chkr1011 — CHANGES_REQUESTED** [review]

**Review by chkr1011 — APPROVED** [review]

**Comment by chkr1011** [comment]
Thank you for your contribution 👍

**Comment by chensiel** [comment]
> From spec:
> Followed by a Byte with a value of either 0 or 1. It is a Protocol Error to include Maximum QoS more than once, or to have a value other than 0 or 1. If the Maximum QoS is absent, the Client uses a Maximum QoS of 2.

Ok thank you.
My broker send a wrong max qos value to me.

**Pull request merged**